gimme

```diff
@@ -100,15 +147,19 @@ _extract() {
 _compile() {
 	(
 		local bootstrap_root=""
-		for bootstrap_root in \
-			"${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_ARCH}" \
-			"${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_HOSTARCH}" ; do
-			if [ -d "${bootstrap_root}" ] ; then
-				break
-			fi
-		done
+		local bootstrap_arch_candidates=(
+			$(ls -1d "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_ARCH}" 2>/dev/null)
+		)
+		local bootstrap_hostarch_candidates=(
+			$(ls -1d "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_HOSTARCH}" 2>/dev/null)
```

Comment:
I should mention that I couldn't get `*` to expand even with only the `GIMME_ARCH` line given, which is why I got silly with `ls -1d`.